docs/migration/cosmoshub-4-v13-upgrade.md

```diff
@@ -0,0 +1,277 @@
+````
+
+build Gaia v13.0.0, and move gaiad v13.0.0 to `$GAIA_HOME/cosmovisor/upgrades/v13/bin`
+
+```shell
+mkdir -p  $GAIA_HOME/cosmovisor/upgrades/v13/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/upgrades/v13/bin
+```
+
+Then you should get the following structure:
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── gaiad  #v12.0.x
+└── upgrades
+    └── v13
+        └── bin
+            └── gaiad  #v13.0.0
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+# please note `DAEMON_HOME` has to be absolute path
+export DAEMON_HOME=$GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Start the node:
+
+```shell
+cosmovisor run  start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
+```
+
+Skipping the invariant checks is strongly encouraged since it decreases the upgrade time significantly and since there are some other improvements coming to the crisis module in the next release of the Cosmos SDK.
+
+#### Expected upgrade result
+
+When the upgrade block height is reached, Gaia will panic and stop:
+
+This may take 7 minutes to a few hours.
+After upgrade, the chain will continue to produce blocks when validators with a total sum voting power > 2/3 complete their node upgrades.
+
+### Auto-Downloading the Gaia binary
+
+**This method is not recommended!**
+
+#### Preparation
+
+Install the latest version of Cosmovisor (`1.5.0`):
+
+```shell
+go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@latest
+```
+
+Create a cosmovisor folder:
+
+create a cosmovisor folder inside gaia home and move gaiad v12.0.x into `$GAIA_HOME/cosmovisor/genesis/bin`
+
+```shell
+mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+```
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+└── genesis
+     └── bin
+        └── gaiad  #v12.0.x
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+export DAEMON_HOME=$GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+export DAEMON_ALLOW_DOWNLOAD_BINARIES=true
+```
+
+Start the node:
+
+```shell
+cosmovisor run start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
+```
+
+Skipping the invariant checks can help decrease the upgrade time significantly.
+
+#### Expected result
+
+When the upgrade block height is reached, you can find the following information in the log:
+
+```shell
+ERR UPGRADE "v13" NEEDED at height: 17380000: upgrade to v13 and applying upgrade "v13" at height:17380000
+```
+
+Then the Cosmovisor will create `$GAIA_HOME/cosmovisor/upgrades/v13/bin` and download the Gaia v13.0.0 binary to this folder according to links in the `--info` field of the upgrade proposal.
+This may take 7 minutes to a few hours, afterwards, the chain will continue to produce blocks once validators with a total sum voting power > 2/3 complete their nodes upgrades.
```

Comment:
7 min is very specific, maybe change to a few mins. 7 min is from estimation of the testnet of theta upgrade.